File: extension3/VisitorGeneratorExt3.py
from __future__ import annotations
from typing import TYPE_CHECKING
from extension3.VisitorExt3 import VisitorExt3
from root.VisitorGenerator import VisitorGenerator

if TYPE_CHECKING:
    from root.CoffeeHouse import CoffeeHouse


class VisitorGeneratorExt3(VisitorGenerator):
    def __init__(self, model: CoffeeHouse):
        super().__init__(model)

    def _generateVisitor(self) -> VisitorExt3:
        return VisitorExt3(self._model)

File: cplusplus/ext3/VisitorGeneratorExt3.cpp
#include "VisitorGeneratorExt3.hpp"

VisitorGeneratorExt3::VisitorGeneratorExt3(CoffeeHouse* model) : VisitorGenerator(model) {
}

Visitor* VisitorGeneratorExt3::generateVisitor() {
	return new VisitorExt3(model);
}
